shapeshift/web
# feat(swapper): add RUNE swaps support

## Description

- [x] Swapper v1 support - working but we might want to revert it from this diff
- [x] Swapper v2 support
- [x] Multi-account support - 35a4f835e2953ebc243fb134957b70a0473b2f9d
- [x] Regression test other swaps - tested Cosmos SDKs, UTXOs, EVM swaps on ZRX/CoW untested yet
- [ ] ~~Swap complete state~~ Tx history is currently borked, so testing the inbound Tx is correctly parsed to set complete state is impossible currently

## Notice




## Pull Request Type


## Issue (if applicable)



closes https://github.com/shapeshift/lib/issues/957

## Risk

Other swappers could be borked WRT passing bip44Params down, regression test accordingly that they work both with the MultiAccounts flag on and off.

## Testing

- Ensure swaps from RUNE work
- Ensure swaps to RUNE work
- With the `MultiAccounts` flag on, ensure selected account other than zero is used for the swap both on the buy and sell side

### Engineering



- Swap logic looks sound
- Thorchain is added in all code paths where we switch on ChainIds / Cosmos ChainNamespace

### Operations



- No SwapperV2 regressions are found

## Screenshots (if applicable)

<img width="408" alt="image" src="https://user-images.githubusercontent.com/17035424/192160669-fb3231b9-b128-4b67-99a8-a7dc09d4a9b2.png">
<img width="400" alt="image" src="https://user-images.githubusercontent.com/17035424/192160681-badcbb0b-37b4-4eed-bcd2-89d3a6800ac4.png">

<img width="400" alt="image" src="https://user-images.githubusercontent.com/17035424/192161060-26eaed84-9768-4cf2-8958-aa2c9e881e57.png">
<img width="402" alt="image" src="https://user-images.githubusercontent.com/17035424/192161068-4235bbe9-356d-4bd2-98be-45a03ebf1670.png">
